sled-agent/src/sim/server.rs

```diff
@@ -87,49 +88,61 @@ impl Server {
+                            sa_address: sa_address.to_string(),
+                            role: NexusTypes::SledRole::Scrimlet,
+                            baseboard: NexusTypes::Baseboard {
+                                serial_number: format!(
+                                    "sim-{}",
+                                    &config.id.to_string()[0..8]
+                                ),
+                                part_number: String::from("Unknown"),
+                                revision: 0,
+                            },
+                            usable_hardware_threads: config
+                                .hardware
+                                .hardware_threads,
+                            usable_physical_ram:
+                                NexusTypes::ByteCount::try_from(
+                                    config.hardware.physical_ram,
+                                )
+                                .unwrap(),
+                            reservoir_size: NexusTypes::ByteCount::try_from(
+                                config.hardware.reservoir_ram,
+                            )
+                            .unwrap(),
                         },
-                        usable_hardware_threads: config
-                            .hardware
-                            .hardware_threads,
-                        usable_physical_ram: NexusTypes::ByteCount::try_from(
-                            config.hardware.physical_ram,
-                        )
-                        .unwrap(),
-                        reservoir_size: NexusTypes::ByteCount::try_from(
-                            config.hardware.reservoir_ram,
-                        )
-                        .unwrap(),
-                    },
-                )
-                .await
-                .map_err(BackoffError::transient)
-        };
-        let log_notification_failure = |error, delay| {
-            warn!(log, "failed to contact nexus, will retry in {:?}", delay;
-                "error" => ?error);
-        };
-        retry_notify(
-            retry_policy_internal_service_aggressive(),
-            notify_nexus,
-            log_notification_failure,
-        )
-        .await
-        .expect("Expected an infinite retry loop contacting Nexus");
+                    )
+                    .await
+                    .map_err(BackoffError::transient)
+            };
+            let log_notification_failure = |error, delay| {
+                warn!(log, "failed to contact nexus, will retry in {:?}", delay;
+                    "error" => ?error);
+            };
+            retry_notify(
+                retry_policy_internal_service_aggressive(),
+                notify_nexus,
+                log_notification_failure,
+            )
+            .await
+            .expect("Expected an infinite retry loop contacting Nexus");
+        });
+
+        if wait_for_nexus {
+            task.await.unwrap();
+        }
```

Comment:
Previously during startup of the simulated sled agent, we always waited to successfully contact Nexus.  That's now up to the caller.  The logic has not otherwise changed here.